src/main/java/co/cask/hydrator/salesforce/SalesforceUtils.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package co.cask.hydrator.salesforce;
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.hydrator.salesforce.authenticator.Authenticator;
+import co.cask.hydrator.salesforce.authenticator.AuthenticatorCredentials;
+import com.sforce.soap.partner.DescribeSObjectResult;
+import com.sforce.soap.partner.Field;
+import com.sforce.soap.partner.FieldType;
+import com.sforce.soap.partner.PartnerConnection;
+import com.sforce.ws.ConnectionException;
+import com.sforce.ws.ConnectorConfig;
+import org.antlr.v4.runtime.ANTLRInputStream;
+import org.antlr.v4.runtime.BailErrorStrategy;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import soql.SOQLLexer;
+import soql.SOQLParser;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * General Salesforce utils
+ */
+public class SalesforceUtils {
+  public static final String API_VERSION = "45.0";
+  public static final String SOAP_API_VERSION = "40.0"; // salesforce soap client does not yet support later versions
+
+  private static final Logger LOG = LoggerFactory.getLogger(SalesforceUtils.class);
+
+  private static SOQLParser.StatementContext getStatementContext(String query) {
+    SOQLLexer lexer = new SOQLLexer(new ANTLRInputStream(query));
+    CommonTokenStream tokens = new CommonTokenStream(lexer);
+    SOQLParser parser = new SOQLParser(tokens);
+    parser.setErrorHandler(new BailErrorStrategy());
+    return parser.statement();
+  }
+
+  /**
+   * Get sObject from SOQL query
+   * Example for "SELECT Id FROM Opportunity" it returns "Opportunity".
+   * For nested queries returns top sObject
+   *
+   * @param query SOQL query
+   *
+   * @return sObject name
+   */
+  public static String getsObjectFromQuery(String query) {
```

Comment:
```suggestion
  public static String getSObjectFromQuery(String query) {
```
Having `getsObject...` returns an impressions that you are using `gets` word.